File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        left=0
        right=len(height)-1
        x=right-left 
        output=[]
        while x>0:
            if height[left]>height[right]:
                area=x*height[right]
                output.append(area)
                right-=1
            elif height[left]<=height[right]:
                area=x*height[left]
                output.append(area)
                left+=1
            x=right-left 
        return max(output)

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int max = 0;

        int i = 0, j = height.length - 1;
        while (i < j) {
            int x = j - i;
            int y = Math.min(height[i], height[j]);
            max = Math.max(max, x * y);

            if (height[i] < height[j]) i++;
            else j--;
        }

        return max;
    }
}